test/functional/wallet_crosschain.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_raises_rpc_error,
+)
+
+class WalletCrossChain(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.setup_clean_chain = True
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def setup_network(self, split=False):
+        self.setup_nodes()
+
+    def run_test(self):
+        self.log.info("Generating initial blockchains")
+        self.nodes[0].generate(5)
+        self.nodes[1].generate(5)
+
+        self.log.info("Creating wallets")
+        node_0_wallet = os.path.join(self.nodes[0].datadir, 'node_0_wallet')
+        self.nodes[0].createwallet(node_0_wallet)
+        self.nodes[0].unloadwallet(node_0_wallet)
+
+        node_1_wallet = os.path.join(self.nodes[1].datadir, 'node_1_wallet')
+        self.nodes[1].createwallet(node_1_wallet)
+        self.nodes[1].unloadwallet(node_1_wallet)
+
+        node_2_wallet = os.path.join(self.nodes[2].datadir, 'node_2_wallet')
+        self.nodes[2].createwallet(node_2_wallet)
+        self.nodes[2].unloadwallet(node_2_wallet)
+
+        self.log.info("Loading wallets into nodes with a different genesis blocks")
+        assert_raises_rpc_error(-4, 'Wallet files should not be reused across chains.', self.nodes[0].loadwallet, node_1_wallet)
```

Comment:
In ece5cd111660005ce71cb8a7092f96c3e084d975 "tests: add tests for cross-chain wallet use prevention"

This assertion is currently not raised. The two wallets are detected as being on the same chain because all regtest nodes use the same genesis block regardless of what comes after.